python/lsst/daf/butler/core/datastore.py

```diff
@@ -1205,3 +1208,122 @@ def set_retrieve_dataset_type_method(self, method: Callable[[str], DatasetType |
         guess dataset location based on its stored dataset type.
         """
         pass
+
+
+class NullDatastore(Datastore):
+    """A datastore that implements the `Datastore` API but always fails when
+    it accepts any request.
+    """
+
+    @classmethod
+    def setConfigRoot(cls, root: str, config: Config, full: Config, overwrite: bool = True) -> None:
+        # Nothing to do. This is not a real Datastore.
+        pass
+
+    def __init__(
+        self,
+        config: Config | ResourcePathExpression | None,
+        bridgeManager: DatastoreRegistryBridgeManager | None,
+        butlerRoot: ResourcePathExpression | None = None,
+    ):
+        # Name ourselves with the timestamp the datastore
+        # was created.
+        self.name = f"{type(self).__name__}@{time.time()}"
```

Comment:
Good point, although since this datastore should never result in a dataset being stored anywhere there's no requirement for it to be a fixed name. The time is really there to let you know when you created that butler and it will appear in stringified output of `Butler`. Whether anyone cares is a question.